src/components/Courses/Courses.jsx

```diff
@@ -17,29 +16,24 @@ const Courses = () => {
   };
 
   const navigate = useNavigate();
+  const dispatch = useDispatch();
 
   const courses = useSelector((state) => state.courses.courses);
-  const dispatch = useDispatch();
 
   useEffect(() => {
-    let endpoints = [
-      'http://localhost:4000/courses/all',
-      'http://localhost:4000/authors/all',
-    ];
-
-    axios
-      .all(endpoints.map((endpoint) => axios.get(endpoint)))
-      .then(
-        axios.spread(({ data: courses }, { data: authors }) => {
-          dispatch(addCourse(courses.result[0]));
-          dispatch(addAuthor(authors.result));
-        })
-      )
-      .catch((error) => {
-        console.error(error.response);
-      });
+    dispatch(getCurrentUser());
+    dispatch(fetchCourses());
+    dispatch(fetchAuthors());
   }, []);
 
+  const user = useSelector((state) => state.courses.user);
+
+  let isAdmin = false;
+
+  if (user.role === 'admin') {
+    isAdmin = true;
+  }
```

Comment:
You can easily combine these items into a single selector:
```suggestion
  const isUserAdmin = useSelector((state) => state.courses.user === 'admin');
```

You don't have to create extra variables, let's update it in this way. To make it reusable, please, create separate selectors function (take a look at my example [here](https://github.com/dotnet-fizzyy/Diploma/blob/master/web/src/redux/selectors/userSelectors.ts#L4))
Also, put, please, `admin` into a constant.